core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,434 @@
+      .interaction.customizationArgs).toEqual(expectedNewCustomizationArgs);
+  });
+
+  it('should save the answer groups', function() {
+    var testStateName = 'Introduction';
+    var newAnswerGroupRuleSpecs = [ruleObjectFactory.createNew('Contains', {
+      x: 'ate'
+    })];
+    var newAnswerGroupOutcome = {
+      param_changes: [],
+      feedback: [
+        '<p>Omnomnom.</p>'
+      ],
+      dest: 'Question 1'
+    };
+    var expectedNewAnswerGroups = [answerGroupObjectFactory
+      .createNew(newAnswerGroupRuleSpecs, newAnswerGroupOutcome, false)];
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveAnswerGroups(
+      testStateName, expectedNewAnswerGroups);
+    expect(explorationStatesService.getState(testStateName)
+      .interaction.answerGroups).toEqual(expectedNewAnswerGroups);
+  });
+
+  it('should save the default outcome', function() {
+    var testStateName = 'Introduction';
+    var expectedNewDefaultOutcome = {
+      dest: 'Introduction',
+      feedback: [
+        '<p>One is wrong bcz many reasons</p>'
+      ]
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveDefaultOutcome(
+      testStateName, expectedNewDefaultOutcome);
+    expect(explorationStatesService.getState(testStateName)
+      .interaction.defaultOutcome).toEqual(expectedNewDefaultOutcome);
+  });
+
+  it('should save the bridge html', function() {
+    var testStateName = 'Introduction';
+    var expectedNewBridgeHtml = '<p> lets move to next Qsn </p>';
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.saveBridgeHtml(
+      testStateName, expectedNewBridgeHtml);
+    var nextState = explorationStatesService
+      .getState(expectedSimpleEditorManagerQuestionList
+        .getNextStateName(testStateName));
+    expect(nextState.content[0].value).toEqual(expectedNewBridgeHtml);
+  });
+
+  it('should add new question', function() {
+    var expectedInteraction = DEFAULT_INTERACTION;
+    var expectedLastStateName = lastStateName;
+    var expectedDefaultOutcome = {
+      dest: expectedLastStateName,
+      feedback: [''],
+      param_changes: []
+    };
+    simpleEditorManagerService.tryToInit();
+    simpleEditorManagerService.addNewQuestion();
+    var state = explorationStatesService.getState(expectedLastStateName);
+    expect(state.interaction.id).toEqual(expectedInteraction.ID);
+    expect(state.interaction.customizationArgs)
+      .toEqual(expectedInteraction.CUSTOMIZATION_ARGS);
+    expect(state.interaction.defaultOutcome).toEqual(expectedDefaultOutcome);
+    var expectedQuestion = questionObjectFactory.create(
+      expectedLastStateName, state.interaction, '');
+    expect(simpleEditorManagerService.getQuestionList()._questions
+      .some(function(question) {
+        // Checks recursively if two objects are equal.
```

Comment:
angular.equals() is working here. I need to see if any of the question belonging to array '_questions' matches expectedQuestion. How can I do that? I thought of two possibilities 1) to write a function that uses for loop and does angular compare with each question and call that function inside expected().  2) to use some().

I chose the 2nd as its cleaner. can u please suggest any other way, may be any jasmine construct?